dokka-integration-tests/gradle/build.gradle.kts

```diff
@@ -1,52 +1,246 @@
  */
+@file:Suppress("UnstableApiUsage")
+
 import dokkabuild.tasks.GitCheckoutTask
-import org.jetbrains.kotlin.gradle.dsl.ExplicitApiMode
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
+import org.gradle.api.tasks.testing.logging.TestLogEvent.FAILED
+import org.gradle.api.tasks.testing.logging.TestLogEvent.SKIPPED
+import org.gradle.language.base.plugins.LifecycleBasePlugin.VERIFICATION_GROUP
+import org.jetbrains.kotlin.gradle.dsl.ExplicitApiMode.Disabled
 
 plugins {
-    id("dokkabuild.test-integration")
+    id("dokkabuild.kotlin-jvm")
+    `jvm-test-suite`
+    `java-test-fixtures`
 }
 
 dependencies {
-    implementation(projects.utilities)
+    api(projects.utilities)
 
-    implementation(kotlin("test-junit5"))
-    implementation(libs.junit.jupiterApi)
-    implementation(libs.junit.jupiterParams)
+    api(libs.jsoup)
 
-    implementation(gradleTestKit())
+    api(libs.kotlin.test)
+    api(libs.junit.jupiterApi)
+    api(libs.junit.jupiterParams)
 
-    implementation(libs.jsoup)
+    api(gradleTestKit())
 }
 
 kotlin {
     // this project only contains test utils and isn't published, so it doesn't matter about explicit API
-    explicitApi = ExplicitApiMode.Disabled
-}
+    explicitApi = Disabled
 
-val templateProjectsDir = layout.projectDirectory.dir("projects")
+    compilerOptions {
+        optIn.add("kotlin.io.path.ExperimentalPathApi")
+    }
+}
 
 val aggregatingProject = gradle.includedBuild("dokka")
+val templateProjectsDir = layout.projectDirectory.dir("projects")
 
-tasks.integrationTest {
+tasks.integrationTestPreparation {
+    // TODO remove this in https://github.com/Kotlin/dokka/pull/3433
     dependsOn(aggregatingProject.task(":publishToMavenLocal"))
-    dependsOn(
-        checkoutKotlinxCoroutines,
-        checkoutKotlinxSerialization,
+}
+
+tasks.withType<Test>().configureEach {
+    dependsOn(tasks.integrationTestPreparation)
+
+    setForkEvery(1)
+    maxHeapSize = "2G"
+    dokkaBuild.integrationTestParallelism.orNull?.let { parallelism ->
+        maxParallelForks = parallelism
+    }
+
+    val useK2 = dokkaBuild.integrationTestUseK2.get()
+
+    useJUnitPlatform {
+        if (useK2) excludeTags("onlyDescriptors", "onlyDescriptorsMPP")
+    }
+
+    systemProperty("org.jetbrains.dokka.experimental.tryK2", useK2)
+    // allow inspecting projects in temporary dirs after a test fails
+    systemProperty(
+        "junit.jupiter.tempdir.cleanup.mode.default",
+        dokkaBuild.isCI.map { isCi -> if (isCi) "ALWAYS" else "ON_SUCCESS" }.get(),
     )
 
     environment("DOKKA_VERSION", project.version)
 
-    inputs.dir(templateProjectsDir)
+    // environment() isn't Provider API compatible yet https://github.com/gradle/gradle/issues/11534
+    dokkaBuild.integrationTestExhaustive.orNull?.let { exhaustive ->
+        environment("isExhaustive", exhaustive)
+    }
+    dokkaBuild.androidSdkDir.orNull?.let { androidSdkDir ->
+        environment("ANDROID_HOME", androidSdkDir)
```

Comment:
Previously it was set with transformation using `invariantSeparatorsPath`, does it work correctly without it?

Reply:
Good catch. Yes, it will work the same, but it should use an invariant path to help with remote Build Cache.